string class_name
float64[] coordinates
float64[] midpoint
float64[] grasp_point1
float64[] grasp_point2